src/error.rs

```diff
@@ -1,13 +1,40 @@
 use std::{
     array::TryFromSliceError,
-    convert::From,
+    convert::{From, Infallible},
+    ffi::NulError,
     num::{ParseIntError, TryFromIntError},
+    str::Utf8Error,
 };
 
+use cosmian_crypto_base::Error as CryptoError;
+use hex::FromHexError;
 use thiserror::Error;
 
 #[derive(Error, Debug, Clone, PartialEq)]
 pub enum FormatErr {
+    //
+    // External errors conversion
+    #[error("{0}")]
+    CryptoError(String),
+    #[error(transparent)]
+    Infallible(Infallible),
+    #[error(transparent)]
+    NulError(NulError),
+    #[error(transparent)]
+    RegexError(#[from] regex::Error),
+    #[error(transparent)]
+    ParseIntError(#[from] ParseIntError),
```

Comment:
maybe this one can be moved to the `ParsingError` at the bottom of this file, what do you think?